backend/routes/SponsorshipFund.js

```diff
@@ -0,0 +1,69 @@
+const router = require("express").Router();
+let SponsorshipFund = require("../models/SponsorshipFund.model");
+
+router.route("/").get((req, res) => {
+  SponsorshipFund.find()
+    .then((SponsorshipFund) => res.json(SponsorshipFund))
+    .catch((err) => res.status(400).json(err));
+});
+
+router.route("/").post((req, res) => {
+  if (!req.body) {
+    res.status(400);
+    throw new Error("Body has missing values");
```

Comment:
"body is missing" is more clear here I believe - we are not doing validation on any of the body parameters, this check is exclusively seeing if it exists

Reply:
also body will always exist, in the case you don't feed it a body, it will be {}. so ensure your check works for an empty object